microprofile.cpp

```diff
@@ -1272,6 +1272,40 @@ void uprintf(const char* fmt, ...)
 #define uprintf(...) do{}while(0)
 #endif
 
+thread_local char g_FloatToStr[64];
+
+#ifdef MICROPROFILE_LOCALE_INDEPENDENT_FLOAT
+#include <locale.h>
+char* MicroProfileDToStr(float value)
+{
+  memset(g_FloatToStr, 0, 64);
+  char tmp[64];
+  memset(tmp, 0, 64);
+  sprintf(tmp, "%f", value);
+  const struct lconv* loc = localeconv();
+  char* p = &tmp[0];
+  char* q = &g_FloatToStr[0];
+  while(*p)
```

Comment:
With initial code thousand separators are left in destination string. It is better to rearrange loop and rethink increment of q pointer like in the code below:
```
while(*p)
{
  if (*p == loc->decimal_point[0])
    *p = '.';
  if (*p != loc->thousands_sep[0])
    *q++ = *p;
  ++p;
}
```